src/tiled/tilesetdock.cpp

```diff
@@ -73,6 +74,21 @@ using namespace Tiled::Internal;
 
 namespace {
 
+class NewTilesetView : public QWidget
+{
+public:
+    explicit NewTilesetView(TilesetDock *parent = nullptr)
+        : QWidget(parent)
+    {
+        QWidget *w = new QWidget(this);
```

Comment:
This widget serves no purpose apart from adding just another level to the widget hierarchy.

Reply:
Alright, I'll take a look at all of that.

I tried to center the button, but everything I tried didn't work.

Reply:
You basically just need spacer items on all 4 sides for centering the button, and for that it needs to be in a grid layout.